tests/cmsis_rtos_v1/threads/thread_apis/src/main.c

```diff
@@ -0,0 +1,136 @@
+	zassert_true(thread1_evidence == 1, "thread1_evidence is not 1\n");
+
+	/* Lower priority */
+	status = osThreadSetPriority(id, osPriorityLow);
+	zassert_true(status == osOK, "Error setting thread priority\n");
+
+	/* Yield to thread of equal or higher priority */
+	status = osThreadYield();
+	zassert_true(status == osOK, "Error doing thread yield\n");
+
+	/* thread1_evidence should now be 2 as it was incremented
+	 * in thread2.
+	 */
+	zassert_true(thread1_evidence == 2, "thread1_evidence is not 2\n");
+}
+
+void thread2(void const *argument)
+{
+	/* By now thread1 would have set thread1_evidence to 1 and would have
+	 * yielded the CPU after lowering its priority. Incrementing it over
+	 * here would essentially confirm that the yield was indeed executed.
+	 */
+	thread1_evidence++;
+
+	/* Increase the priority of thread1 to ensure it gets scheduled */
+	osThreadSetPriority(id1, osPriorityHigh);
+}
+
+void thread3(void const *argument)
+{
+	osPriority rv;
+	osThreadId id = osThreadGetId();
+	osPriority prio = osThreadGetPriority(id);
+
+	/* Lower the priority of the current thread */
+	osThreadSetPriority(id, osPriorityBelowNormal);
+	rv = osThreadGetPriority(id);
+	zassert_equal(rv, osPriorityBelowNormal,
+			"Expected priority to be changed to %d, not %d\n",
+			(int)osPriorityBelowNormal, (int)rv);
+
+	/* Increase the priority of the current thread */
+	osThreadSetPriority(id, osPriorityAboveNormal);
+	rv = osThreadGetPriority(id);
+	zassert_equal(rv, osPriorityAboveNormal,
+			"Expected priority to be changed to %d, not %d\n",
+			(int)osPriorityAboveNormal, (int)rv);
+
+	/* Restore the priority of the current thread */
+	osThreadSetPriority(id, prio);
+	rv = osThreadGetPriority(id);
+	zassert_equal(rv, prio,
+			"Expected priority to be changed to %d, not %d\n",
+			(int)prio, (int)rv);
+
+	/* Indication that thread3 is done with its processing */
+	thread3_state = 1;
+
+	/* Keep looping till it gets killed */
+	do {
+		osDelay(100);
+	} while (1);
+}
+
+osThreadDef(thread1, osPriorityHigh, 1, STACKSZ);
+osThreadDef(thread2, osPriorityHigh, 1, STACKSZ);
+osThreadDef(thread3, osPriorityNormal, 1, STACKSZ);
+
+void test_cmsisthread_prio(void)
+{
+	osStatus status;
+	osThreadId id3;
+
+	id3 = osThreadCreate(osThread(thread3), NULL);
+	zassert_true(id3 != NULL, "Failed creating thread3\n");
+
+	/* Keep delaying 100 millisecond to ensure thread3 is done with
+	 * its execution. It loops at the end and is terminated here.
+	 */
+	do {
+		status = osDelay(10);
+	} while (thread3_state == 0);
+
+	status = osThreadTerminate(id3);
+	zassert_true(status == osOK, "Error terminating thread3\n");
+	thread3_state = 0;
+}
+
+void test_cmsisthread_apis(void)
+{
+	osThreadId id2;
+	osStatus status;
+
+	id1 = osThreadCreate(osThread(thread1), NULL);
+	zassert_true(id1 != NULL, "Failed creating thread1\n");
+
+	id2 = osThreadCreate(osThread(thread2), NULL);
+	zassert_true(id2 != NULL, "Failed creating thread2\n");
+
+	do {
+		status = osDelay(100);
```

Comment:
Did you mean to make use of status variable below? If not, can remove

Reply:
yes. will remove.